specification.md

```diff
@@ -115,11 +115,13 @@ const ListModule2 = {
   }
 }
 
-const list = {'fantasy-land/canonical': ListModule2, data: [1]}
+const list = {'fantasy-land/canonical': () => ListModule2, data: [1]}
 ```
 
 Note that the `ListModule2` here is correct. Only the `list` value doesn't follow the specification.
 
+The `fantasy-land/canonical` method must always return equivalent modules when called multiple times.
```

Comment:
I don't quite understand what's the point of `fantasy-land/canonical` being a function. like if we ask it always return equivalent modules when called multiple times, i.e. being pure, then it's only advantage I see is being able to lazily contract the value, which i don't see as significant optimization, I also haven't read that conversation so all this might be discussed in that issue

Reply:
I think the discussion boils down to this sentence:

> By not making it a thunk the API is imposing an evaluation strategy, which in JavaScript in particular is a pretty serious problem due to the order of evaluation. (c) @alexandru

I'm also not sure about the importance of such flexibility, but on the other hand seems like it won't hurt anyway.

Reply:
@alexandru can you point out "practical" example where fantasy-land/canonical being lazy is more useful/better then it being non lazy?